### PR TITLE
Fix collection bug

### PR DESCRIPTION
**Pull request by dmitrykruglov**

### CODE REVIEW

**Review by masha1997 — APPROVED** [review]

File: provider/resource_collection.go [hunk 1]
```diff
@@ -501,7 +501,7 @@ func collectionApiEntriesDeleteOldRows(ctx context.Context, d *schema.ResourceDa
 
 	gen := api.Data["generation"].(int)
 	scope := api.Data["scope"].(string)
-	q := fmt.Sprintf("{\"$or\":[{\"_gen\":null},{\"_gen\":{\"$lt\":\"%d\"}}]}", gen)
+	q := fmt.Sprintf("{\"$or\":[{\"_gen\":null},{\"_gen\":{\"$ne\": %d}}]}", gen)
```
Comment (masha1997) [inline provider/resource_collection.go:504]: nice

Comment (masha1997) [reply provider/resource_collection.go:504]: If we end up with _gen=9 & _gen=12. what is guarantee us that 9 ones will be deleted?

**Review by benoittoulme — APPROVED** [review]
Do we need to do anything to the data when updating?

File: provider/resource_collection.go (hunk 1 above)
Comment (dmitrykruglov) [reply provider/resource_collection.go:504]: this will remove all but the current _gen. If current _gen in 12, it will delete everything else

Comment (masha1997) [reply provider/resource_collection.go:504]: thank you

**Comment by dmitrykruglov** [comment]
> Do we need to do anything to the data when updating?

No, should be backward compatible

**Pull request merged**